src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,47 +3,171 @@
+    
+
+    @FXML
+    private VBox mainLayout;
+    @FXML
+    private Label title;
+    @FXML
+    private FlowPane assignedLabels;
+    @FXML
+    private TextField queryField;
+    @FXML
+    private VBox feedbackLabels;
 
     LabelPickerDialog(TurboIssue issue, List<TurboLabel> repoLabels, Stage stage) {
+        this.repoLabels = repoLabels;
+        this.repoLabelsString = repoLabels.stream()
+                .map(TurboLabel::getActualName)
+                .collect(Collectors.toSet());
+        this.issue = issue;
+        LabelPickerState state = new LabelPickerState(new HashSet<String>(issue.getLabels()));
+        uiLogic = new LabelPickerUILogic();
+
+        // Logic initialisation
+        this.groups = initGroups(repoLabels);
+
         // UI creation
+        initUI(stage, issue);
+        updateUI(state);
+        setupEvents(stage);
+        Platform.runLater(queryField::requestFocus);
+    }
+
+    // Init Assets 
+    private Map<String, Boolean> initGroups(List<TurboLabel> repoLabels) {
+        Map<String, Boolean> groups = new HashMap<>();
+        repoLabels.forEach(label -> {
+            if (label.getGroup().isPresent() && !groups.containsKey(label.getGroup().get())) {
+                groups.put(label.getGroup().get(), label.isExclusive());
+            }        
+        });
+        return groups;
+    }
+
+    // Initilisation of UI
+
+    private void initialiseDialog(Stage stage, TurboIssue issue) {
+        initOwner(stage);
+        initModality(Modality.APPLICATION_MODAL); // TODO change to NONE for multiple dialogs
+        setTitle("Edit Labels for " + (issue.isPullRequest() ? "PR #" : "Issue #") +
+                issue.getId() + " in " + issue.getRepoId());
+    }
+
+    private void positionDialog(Stage stage) {
+        if (!Double.isNaN(getHeight())) {
+            setX(stage.getX() + stage.getScene().getX());
+            setY(stage.getY() +
+                 stage.getScene().getY() +
+                 (stage.getScene().getHeight() - getHeight()) / 2);
+        }
+    }
+
+
+    private void initUI(Stage stage, TurboIssue issue) {
         initialiseDialog(stage, issue);
         createButtons();
-        VBox vBox = createVBox();
-        Label titleLabel = createTitleLabel(issue);
-        titleLabel.setTooltip(createTitleTooltip(issue));
-        topPane = createTopPane();
-        textField = createTextField();
-        bottomBox = createBottomBox();
+        setDialogPaneContent(issue);
+        title.setTooltip(createTitleTooltip(issue));
+    }
 
-        setupEvents(stage);
-        uiLogic = new LabelPickerUILogic(issue, repoLabels, this);
+    private void setDialogPaneContent(TurboIssue issue) {
+        try {
+            createMainLayout();
+            setTitleLabel(issue);
+            getDialogPane().setContent(mainLayout);
+        } catch (IOException e) {
+            // TODO use a HTLogger instead when failed to load fxml
+            e.printStackTrace();
```

Comment:
Does anything else need to be done here to recover from the error? Maybe we should close the dialog instead of showing a partially-loaded interface.